caikit_nlp/resources/pretrained_model/hf_auto_causal_lm.py

```diff
@@ -198,69 +270,123 @@ def _force_to_batch_encoding_list(
+            HFAutoCausalLM._concatenate_encodings(source_ids, target_ids)
+            chunks = HFAutoCausalLM._split_encoding_into_chunks(
+                encoding=source_ids,
+                chunk_size=chunk_size,
+                drop_remainder=drop_remainder,
+                task_ids=task_ids,
+            )
+            return chunks
         # Otherwise we need to expand the dict along its keys,
         # mapping all of its encapsulated objects to new items.
         encodings = []
-        num_target_samples = []
         id_keys = source_ids.keys()
         key = None
         error.value_check(
             "<NLP94411004E>",
             source_ids.keys(),
             "Source ID batch encoding must have keys",
         )
+
         for batch_idx in range(len(source_ids.input_ids)):
             new_encoding = BatchEncoding()
             for key in id_keys:
-                if key == "input_ids":
-                    new_encoding[key] = (
-                        source_ids[key][batch_idx] + target_ids[key][batch_idx]
-                    )
-                else:
-                    new_encoding[key] = source_ids[key][batch_idx]
-            num_target_samples.append(len(target_ids[key][batch_idx]))
-            new_encoding["task_ids"] = task_ids
-            encodings.append(new_encoding)
-        return encodings, num_target_samples
+                new_encoding[key] = (
+                    source_ids[key][batch_idx] + target_ids[key][batch_idx]
+                )
+            chunks = HFAutoCausalLM._split_encoding_into_chunks(
+                encoding=new_encoding,
+                chunk_size=chunk_size,
+                drop_remainder=drop_remainder,
+                task_ids=task_ids,
+            )
+            # Chunks are held as a list of lists
+            encodings += chunks
+        return encodings
 
     @staticmethod
-    def _get_attention_mask(
-        source_ids: BatchEncoding, idx: int, num_target_samples: int
-    ) -> List[int]:
-        """Get the attention mask for a given target token from some source encoding.
+    def _concatenate_encodings(left: BatchEncoding, right: BatchEncoding) -> None:
+        """Given two batch encodings, combine their entries into a single encoding.
 
         Args:
-            source_ids: BatchEncoding
-                Source encoding that requires an attention mask.
-            idx: int
-                Index of the output token we attend up to.
-            num_target_samples: int
-                Length of the original target seequence being considered.
+            left: BatchEncoding
+                Encoding representing left sequence, which will be updated in place.
+                Corresponds to source.
+            right: BatchEncoding
+                Encoding representing right sequence, which will be stacked onto the left
+                encoding. Corresponds to target.
+        """
+        for k in left.keys():
```

Comment:
since this will get called a lot of time, shall we use dictionary comprehension here?